src/app/server/Dnssd.cpp

```diff
@@ -293,7 +300,14 @@ CHIP_ERROR DnssdServer::Advertise(bool commissionableNode, chip::Dnssd::Commissi
 
     uint8_t macBuffer[DeviceLayer::ConfigurationManager::kPrimaryMACAddressLength];
     MutableByteSpan mac(macBuffer);
-    chip::DeviceLayer::ConfigurationMgr().GetPrimaryMACAddress(mac);
+    if (chip::DeviceLayer::ConfigurationMgr().GetPrimaryMACAddress(mac) != CHIP_NO_ERROR)
```

Comment:
This should never fail. Perhaps a more root cause fix is to make all paltforms do something sane if they can't access the interface's mac address. Perhaps the network interfaces are down, and it will just die further down.

Reply:
The method is declared returning CHIP_ERROR and apparently it can fail given our implementation.

Changing it so it never fails seems outside the scope of this PR. I am not sure how to even fix that on non-embedded:
  - returning a hardcoded value was wrong
  - I think mac may need indexing, so if I return a random value, I should probably ensure that value is stable at least while the instance is running.
  
I believe a better fix would be 'random if cannot get, persist that random somewhere (kvs?)' but that change stars to veer of into discussions & complexity. We need a fix sooner than that.

Reply:
I also 100% agree we should not die in this particular situation: DNSSD requires some ip interface active, in which case a mac address is expected.

However I do not feel as confident in doing `verifyOrDie( == CHIP_NO_ERROR)`. Maybe someone more confident can follow up with a PR that does that and explain reasoning. For now I believe random mac will work for DNS purposes somewhat ... not ideal for sure because I believe we do not want our host name to change too often.